python_modules/dagster/dagster/_core/definitions/asset_condition/scheduling_condition_evaluation_context.py

```diff
@@ -0,0 +1,114 @@
+import dataclasses
+import datetime
+import functools
+import logging
+from dataclasses import dataclass
+from typing import AbstractSet, Mapping, Optional, Tuple
+
+import pendulum
+
+from dagster._core.asset_graph_view.asset_graph_view import (
+    AssetGraphView,
+    AssetSlice,
+)
+from dagster._core.definitions.asset_condition.asset_condition import (
+    AssetCondition,
+    AssetConditionEvaluation,
+    AssetConditionEvaluationState,
+)
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_subset import ValidAssetSubset
+from dagster._core.definitions.events import AssetKeyPartitionKey
+
+from .asset_condition_evaluation_context import AssetConditionEvaluationContext
+
+
+@dataclass(frozen=True)
+class SchedulingConditionEvaluationContext:
+    # the AssetKey of the currently-evaluated asset
+    asset_key: AssetKey
+
+    # the condition that is being evaluated
+    condition: AssetCondition
+    # the subset of AssetPartitions for this Asset which must be evaluated. at the root of the
+    # condition evaluation tree, this is the AllPartitionsSubset, but this may shrink as conditions
+    # are applied and we can be certain that certain partitions will not be true for the overall
+    # expression
+    candidate_subset: ValidAssetSubset
```

Comment:
let's do `AssetSlice`.

Also I think we need to think asset check scheduling here.

Reply:
This is changed by the end of this stack so letting it go